Tooltip of favorites icon does not appear

**Expected:**
Tooltip on-hover as specified in #4103
![bildschirmfoto 2018-03-13 um 09 41 29](https://user-images.githubusercontent.com/7469/37957450-39f0af2a-31ae-11e8-9e42-c5416e6519dc.png)

**Got:**
Tooltip does not appear.

Sehr schön:
```
Failure in test test_adds_is_favorite_class_when_favorite_exists (opengever.base.tests.test_favorite_action.TestFavoriteAction)
Traceback (most recent call last):
  File "/usr/local/python/2.7.14/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/var/lib/jenkins/zope/eggs/ftw.testbrowser-1.29.7-py2.7.egg/ftw/testbrowser/__init__.py", line 42, in test_function
    return func(self, *args, **kwargs)
  File "/var/lib/jenkins/jobs/opengever.core-weight-5/workspace/opengever/base/tests/test_favorite_action.py", line 59, in test_adds_is_favorite_class_when_favorite_exists
    '', browser.css('#mark-as-favorite').first.get('class'))
  File "/usr/local/python/2.7.14/lib/python2.7/unittest/case.py", line 513, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/local/python/2.7.14/lib/python2.7/unittest/case.py", line 506, in _baseAssertEqual
    raise self.failureException(msg)
AssertionError: '' != None
```
Das passiert, das neu der Attribut nicht gesetzt wird wenn das Objekt kein Favorit ist.
Ich schlage vor, hier den Test anzupassen.

--- a/docs/HISTORY.txt
+++ b/docs/HISTORY.txt
@@ -8,6 +8,7 @@ Changelog
 - Omit parentheses if no abbreviation for directorate or department available. [tarnap]
 - Allow attaching documents of sibling proposals to proposals. [Rotonen]
 - Add today to the data passed to the sablon template. [tarnap]
+- Display a tooltip for the favorite action. [tarnap]
 - Make document mimetype lookups case insensitive. [Rotonen]
 - Reset task responsible to the task issuer upon task rejection. [Rotonen]
 - Do not apply the current date to meeting titles. [Rotonen]

--- a/opengever/base/browser/resources/Favorites.js
+++ b/opengever/base/browser/resources/Favorites.js
@@ -13,8 +13,10 @@
       }
 
       if ($('#mark-as-favorite').hasClass('is-favorite')){
+        target.attr('title', target.data('markFavoriteTitle'));
         this.delete_favorite(target);
       } else {
+        target.attr('title', target.data('unmarkFavoriteTitle'));
         this.add_favorite(target);
       }
     };

--- a/opengever/base/locales/de/LC_MESSAGES/opengever.base.po
+++ b/opengever/base/locales/de/LC_MESSAGES/opengever.base.po
@@ -4,7 +4,7 @@
 msgid ""
 msgstr ""
 "Project-Id-Version: PACKAGE VERSION\n"
-"POT-Creation-Date: 2018-05-23 15:08+0000\n"
+"POT-Creation-Date: 2018-05-28 15:47+0000\n"
 "PO-Revision-Date: 2014-09-04 01:03+0200\n"
 "Last-Translator: Jonas Baumann <j.baumann@4teamwork.ch>\n"
 "Language-Team: Jonas Baumann <j.baumann@4teamwork.ch>\n"
@@ -437,6 +437,16 @@ msgstr "Benutzer-Aktion bestätigen"
 #: ./opengever/base/browser/templates/favorites.pt
 msgid "title_manage_favorites"
 msgstr "Favoriten"
+
+#. Default: "Mark as favorite"
+#: ./opengever/base/viewlets/favorite_action.pt
+msgid "tooltip_mark_favorite_action"
+msgstr "Als Favorit markieren"
+
+#. Default: "Remove from favorites"
+#: ./opengever/base/viewlets/favorite_action.pt
+msgid "tooltip_remove_favorite_action"
+msgstr "Aus Favoriten entfernen"
 
 #. Default: "You are not allowed to see this reference"
 #: ./opengever/base/browser/folder_delete_confirmation.py

--- a/opengever/base/locales/fr/LC_MESSAGES/opengever.base.po
+++ b/opengever/base/locales/fr/LC_MESSAGES/opengever.base.po
@@ -4,7 +4,7 @@
 msgid ""
 msgstr ""
 "Project-Id-Version: PACKAGE VERSION\n"
-"POT-Creation-Date: 2018-05-23 15:08+0000\n"
+"POT-Creation-Date: 2018-05-28 15:47+0000\n"
 "PO-Revision-Date: 2017-11-29 10:53+0000\n"
 "Last-Translator: Philippe Gross <gross.philippe@gmail.com>\n"
 "Language-Team: French <https://translations.onegovgever.ch/projects/onegov-gever/opengever-base/fr/>\n"
@@ -436,6 +436,16 @@ msgstr "Confirmation de l'action"
 #: ./opengever/base/browser/templates/favorites.pt
 msgid "title_manage_favorites"
 msgstr "Favoris"
+
+#. Default: "Mark as favorite"
+#: ./opengever/base/viewlets/favorite_action.pt
+msgid "tooltip_mark_favorite_action"
+msgstr ""
+
+#. Default: "Remove from favorites"
+#: ./opengever/base/viewlets/favorite_action.pt
+msgid "tooltip_remove_favorite_action"
+msgstr ""
 
 #. Default: "You are not allowed to see this reference"
 #: ./opengever/base/browser/folder_delete_confirmation.py

--- a/opengever/base/locales/opengever.base.pot
+++ b/opengever/base/locales/opengever.base.pot
@@ -4,7 +4,7 @@
 msgid ""
 msgstr ""
 "Project-Id-Version: PACKAGE VERSION\n"
-"POT-Creation-Date: 2018-05-23 15:08+0000\n"
+"POT-Creation-Date: 2018-05-28 15:47+0000\n"
 "PO-Revision-Date: YEAR-MO-DA HO:MI +ZONE\n"
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
@@ -432,6 +432,15 @@ msgstr ""
 #. Default: "Favorites"
 #: ./opengever/base/browser/templates/favorites.pt
 msgid "title_manage_favorites"
+msgstr ""
+
+#. Default: "Mark as favorite"
+#: ./opengever/base/viewlets/favorite_action.pt
+msgid "tooltip_mark_favorite_action"
+msgstr ""
+
+#: ./opengever/base/viewlets/favorite_action.pt
+msgid "tooltip_remove_favorite_action"
 msgstr ""
 
 #. Default: "You are not allowed to see this reference"

--- a/opengever/base/tests/test_favorite_action.py
+++ b/opengever/base/tests/test_favorite_action.py
@@ -56,7 +56,7 @@ class TestFavoriteAction(IntegrationTestCase):
 
         browser.open(self.document)
         self.assertEqual(
-            '', browser.css('#mark-as-favorite').first.get('class'))
+            None, browser.css('#mark-as-favorite').first.get('class'))
 
 
 class TestFavoriteEtagValue(IntegrationTestCase):

--- a/opengever/base/viewlets/favorite_action.pt
+++ b/opengever/base/viewlets/favorite_action.pt
@@ -1,11 +1,17 @@
 <div id="favorite-action"
+     i18n:domain="opengever.base"
      tal:condition="view/available"
      tal:define="favorite view/get_favorite"
      tal:attributes="data-oguid python: view.oguid().id;
                      data-url view/url">
 
-  <a href="#" id="mark-as-favorite"
-     tal:attributes="class python:'is-favorite' if favorite else '';
-                     data-favorite-id python: favorite and favorite.favorite_id" />
+  <a href="#"
+     id="mark-as-favorite"
+     class="is-favorite"
+     tal:attributes="title python: view.get_unmark_favorite_title() if favorite else view.get_mark_favorite_title();
+                     class python: 'is-favorite' if favorite else None;
+                     data-mark-favorite-title view/get_mark_favorite_title;
+                     data-unmark-favorite-title view/get_unmark_favorite_title;
+                     data-favorite-id python: favorite.favorite_id if favorite else ''" />
 
 </div>

--- a/opengever/base/viewlets/favorite_action.py
+++ b/opengever/base/viewlets/favorite_action.py
@@ -1,3 +1,4 @@
+from opengever.base import _
 from opengever.base import is_favorites_feature_enabled
 from opengever.base.favorite import FavoriteManager
 from opengever.base.oguid import Oguid
@@ -19,6 +20,14 @@ class FavoriteAction(common.ViewletBase):
     def get_favorite(self):
         return FavoriteManager().get_favorite(
             self.context, api.user.get_current())
+
+    def get_mark_favorite_title(self):
+        return _('tooltip_mark_favorite_action',
+                 default=u'Mark as favorite')
+
+    def get_unmark_favorite_title(self):
+        return _('tooltip_remove_favorite_action',
+                 default=u'Remove from favorites')
 
     def oguid(self):
         return Oguid.for_object(self.context)

